### PR TITLE
Update dedicated server (and (me_)_installation_template ) 

### PR DESCRIPTION
**Pull request by bigbigbang**
# Description

Update of dedicated_me_installation_template
 - remove deprecated attributes
 - add attributes (end_of_install, subfamily, no_partitioning)

Update of type_installation_template

Add dedicated_installation_template data source

Remove resource me_ssh_key(s)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (improve existing resource(s) or datasource(s))
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

make testacc TESTARGS="-run TestAccInstallationTemplateDataSource_basic"
- [x] Test A: `make testacc TESTARGS="-run TestAccMeInstallationTemplatesDataSource_basic"`
- [x] Test B: `make testacc TESTARGS="-run TestAccInstallationTemplateDataSource_basic"`
- [x] Test C: `make testacc TESTARGS="-run TestAccMeInstallationTemplateResource_basic"`

**Test Configuration**:
* Terraform version: `terraform version`: Terraform 1.6.6
* Existing HCL configuration you used: 
```hcl
data "ovh_dedicated_server" "server" {
  service_name = "nsxxxxx.ip-123-123-123.eu"
}

data "ovh_dedicated_installation_template" "ovhtemplate" {
  template_name = "win2019-dc_64"
}

output "template" {
  value = data.ovh_dedicated_installation_template.ovhtemplate
}
```

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings or issues
- [ ] I have added acceptance tests that prove my fix is effective or that my feature works
- [ ] New and existing acceptance tests pass locally with my changes
- [ ] I ran succesfully `go mod vendor` if I added or modify `go.mod` file

### CODE REVIEW

**Review by amstuta — CHANGES_REQUESTED** [review]

File: ovh/data_dedicated_installation_template.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ovh/go-ovh/ovh"
+	"github.com/ovh/terraform-provider-ovh/ovh/helpers"
+)
+
+func dataSourceDedicatedInstallationTemplate() *schema.Resource {
```
[the remark this answers is not in the document]
Comment (bigbigbang) [reply ovh/data_dedicated_installation_template.go:12]: no, it isn't the same API call ; one is me_installation_template -> return customer custom templates, and the other call dedicated_installation_template -> return OVH template

File: website/docs/d/dedicated_installation_template.html.markdown [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## Argument Reference
+
+* `template_name` - (Required) The name of the template
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `id` - The ID of the installation template.
+* `bit_format`: Template bit format (32 or 64).
```
Comment (amstuta) [inline website/docs/d/dedicated_installation_template.html.markdown:30]:
```suggestion
* `bit_format`: Template bit format (32 or 64)
```

(to remove or add on all fields to be consistent)

File: ovh/data_dedicated_installation_template_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package ovh
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccInstallationTemplateDataSource_basic(t *testing.T) {
```
Comment (amstuta) [inline ovh/data_dedicated_installation_template_test.go:10]:
```suggestion
func TestAccDedicatedInstallationTemplateDataSource_basic(t *testing.T) {
```

File: .cds/terraform-provider-ovh.yml [hunk 4]
```diff
@@ -544,14 +544,14 @@ workflow:
...
-  Tests_TestAccDedicatedInstallationTemplates:
+  Tests_TestAccDedicatedInstallationTemplate:
     application: terraform-provider-ovh
     depends_on:
     - terraform-provider-ovh-pre-sweepers
     environment: acctests
     one_at_a_time: true
     parameters:
-      testargs: -run TestAccDedicatedInstallationTemplates
+      testargs: -run TestAccDedicatedInstallationTemplateDataSource_basic
```
Comment (amstuta) [inline .cds/terraform-provider-ovh.yml:554]:
```suggestion
      testargs: -run TestAccDedicatedInstallationTemplate
```

(we need to run `TestAccDedicatedInstallationTemplatesDataSource_basic` also)

File: ovh/data_dedicated_installation_template_test.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	testAccInstallationTemplateDatasourceConfig_Basic := `data "ovh_dedicated_installation_template" "template" {
+		template_name= "debian12_64"
+	}`
+	config := testAccInstallationTemplateDatasourceConfig_Basic
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheckCredentials(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
```
Comment (amstuta) [inline ovh/data_dedicated_installation_template_test.go:23]:
```suggestion
				Config: testAccInstallationTemplateDatasourceConfig_Basic,
```

and you can remove variable `config`

**Review by amstuta — APPROVED** [review]

**Pull request merged**